parastell/invessel_build.py

```diff
@@ -767,92 +768,64 @@ def mesh_components_gmsh(
 
-        gmsh.option.setNumber("Mesh.MeshSizeMin", min_mesh_size)
-        gmsh.option.setNumber("Mesh.MeshSizeMax", max_mesh_size)
-
-        gmsh.model.mesh.generate(dim=3)
-
-    def _gmsh_from_pydagmc(self, components):
+    def _gmsh_from_pydagmc(self, components, min_mesh_size, max_mesh_size):
         """Adds PyDAGMC geometry to Gmsh instance.
         (Internal function not intended to be called externally)
 
         Arguments:
             components (array of str): array containing the names of the
                 in-vessel components to be meshed.
+            min_mesh_size (float): minimum size of mesh elements (defaults to
+                5.0).
+            max_mesh_size (float): maximum size of mesh elements (defaults to
+                20.0).
         """
-        # These parameters are taken directly from Gmsh documentation (see
-        # tutorial t13)
-        gmsh.onelab.set(
-            """[
-            {
-                "type": "number",
-                "name": "Parameters/Angle for surface detection",
-                "values": [40],
-                "min": 20,
-                "max": 120,
-                "step": 1
-            },
-            {
-                "type": "number",
-                "name": "Parameters/Create surfaces guaranteed to be parametrizable",
-                "values": [0],
-                "choices": [0, 1]
-            }
-            ]"""
-        )
+        mesh_files = []
 
+        # Extract each component from PyDAGMC model and remesh it in Gmsh
         for component in components:
             volume_id = self.radial_build.radial_build[component]["vol_id"]
 
             vtk_path = str(Path(f"volume_{volume_id}_tmp").with_suffix(".vtk"))
             self.dag_model.volumes_by_id[volume_id].to_vtk(vtk_path)
 
-            gmsh.merge(vtk_path)
-
-        angle = gmsh.onelab.getNumber(
-            "Parameters/Angle for surface detection"
-        )[0]
-        includeBoundary = True
-        forceParameterizablePatches = gmsh.onelab.getNumber(
-            "Parameters/Create surfaces guaranteed to be parametrizable"
-        )[0]
-        curveAngle = 180.0
-
-        gmsh.model.mesh.classifySurfaces(
-            np.deg2rad(angle),
-            includeBoundary,
-            forceParameterizablePatches,
-            np.deg2rad(curveAngle),
-        )
-
-        gmsh.model.mesh.create_geometry()
+            mesh_files.append(
+                remesh_gmsh(min_mesh_size, max_mesh_size, vtk_path)
+            )
 
-        surfaces = gmsh.model.getEntities(dim=2)
-        surface_tags = [s[1] for s in surfaces]
-        surface_loop = gmsh.model.geo.addSurfaceLoop(surface_tags)
-        gmsh.model.geo.addVolume([surface_loop])
+        # Combine all component meshes into one
+        [gmsh.merge(mesh_file) for mesh_file in mesh_files]
 
-        gmsh.model.geo.synchronize()
+        [Path(mesh_file).unlink() for mesh_file in mesh_files]
```

Comment:
Probably simpler to just put this in a more traditional loop with the previous one....